%YAML:1.0

camera:
  type: "PINHOLE"

  width: 752
  height: 480

  fx: 458.654
  fy: 457.296
  cx: 367.215
  cy: 248.375

  k1: -0.28340811
  k2: 0.07395907
  d1: 0.00019359
  d2: 1.76187114e-05

Tbc:
  Rbc: !!opencv-matrix
    rows: 3
    cols: 3
    dt: d
    data: [0.0148655429818, -0.999880929698, 0.00414029679422,
           0.999557249008, 0.0149672133247, 0.025715529948, 
           -0.0257744366974, 0.00375618835797, 0.999660727178]
  
  tbc: !!opencv-matrix
    rows: 3
    cols: 1
    dt: d
    data: [-0.0216401454975,-0.064676986768, 0.00981073058949]

# track parameters
fm_threshold: 1.0
max_cornor_num: 150
min_cornor_gap: 30
track_frequency: 20 
track_verbose: 0

# inertial parameters
inertial_init_cnt: 200
inertial_init_accl_cov: 1.0
inertial_init_gyro_cov: 0.1

# noise parameters
noise_accl: 0.0025
noise_gyro: 0.000025
noise_accl_bias: 0.0001
noise_gyro_bias: 0.000001
noise_observation: 0.001225

# feature optimization
feature:
  max_iter_cnt: 20
  max_try_cnt: 20
  converge_threshold: 5.e-7
  min_disparity_angle: 0 # in deg
  min_disparity_distance: 0.1

# image noise
sliding_window_lens: 20
distance_threshold: 0.5
angle_threshold: 15 # in deg